onnxruntime/core/providers/snpe/snpe_lib.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include "DlSystem/DlError.hpp"
+#include "core/providers/snpe/snpe_runtime_options.h"
+#include "core/common/status.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace snpe {
+
+struct UserBufferAttribute {
+ public:
+  UserBufferAttribute(size_t size,
+                      const std::vector<size_t>& buffer_strides,
+                      zdl::DlSystem::UserBufferEncoding* const buffer_encoding) :
+      buffer_size(size),
+      strides(buffer_strides),
+      user_buffer_encoding(buffer_encoding)
+  {}
+
+  size_t buffer_size;
+  std::vector<size_t> strides;
+  zdl::DlSystem::UserBufferEncoding* user_buffer_encoding;
+};
+
+using namespace onnxruntime::common;
```

Comment:
```suggestion
using onnxruntime::common::Status;
```
to avoid including all namespace names in a header